src/main/java/omero/gateway/Gateway.java

```diff
@@ -1135,8 +1134,7 @@ private ExperimenterData login(client client, LoginCredentials cred)
             ServiceFactoryPrx entryEncrypted = client.getSession();
             
             // version check
-            ResourceBundle bundle = ResourceBundle.getBundle("omero");
-            String clientVersion = bundle.getString("omero.version");
+            String clientVersion = BuildConfig.getBuildVersion();
```

Comment:
Since the `BuildConfig` class is only introspecting the package, any reason we couldn't use `self.getClass().getPackage().getImplementationVersion();`

Reply:
I opted to stay with the BuildConfig class for now as it keeps a single source of truth. We can swap out the implementation of how the version is obtained, but not how to change the calling code. 

BuildConfig is also a pattern used on [Android](https://developer.android.com/studio/build/gradle-tips#simplify-app-development) and works very well with Gradle. It's hard to vouch for it, but it's something that has stood up well as the build system became more dynamic.